### PR TITLE
Winston/sobjects

### PR DESCRIPTION
**Pull request by chowwinston**
### What does this PR do?
Fetches and describes each sobjects with integration tests
### What issues does this PR fix or reference?
@W-4252302@

### CODE REVIEW

**Comment by chowwinston** [comment]
This is still a work in progress as I clean up and add more tests. I'm creating this PR first to show how integration tests can be done.

File: packages/salesforcedx-sobjects/test/integration/sobjectDescribe.test.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    label: 'Sample Object',
+    pluralLabel: 'Sample Object',
+    nameField: {
+      type: 'Text',
+      label: 'Sample Object'
+    },
+    deploymentStatus: 'Deployed',
+    sharingModel: 'ReadWrite'
+  };
+  const mdapi = (conn as any).metadata;
```
Comment (chowwinston) [inline packages/salesforcedx-sobjects/test/integration/sobjectDescribe.test.ts]: I'm not sure why metadata is exposed in jsforce.connection

**Comment by vazexqi** [comment]
@chowwinston - Is it a lot of work to rebase onto develop? I am seeing a lot of unrelated changes in this CL that makes it hard to get to the gist of things. This might also make it hard for @muenzpraeger, @ayeshakmaz and @engai  if they are looking at this PR as a way to do the integration tests.

**Comment by chowwinston** [comment]
Did the rebase. Not sure why some diffs are still showing up, but the ones from the refactor are gone.

**Comment by vazexqi** [comment]
@chowwinston - I think you rebased wrongly. I now see even more commits than before (see https://github.com/forcedotcom/salesforcedx-vscode/pull/92/commits). Let's go through this on Tuesday. It's at the point where it might be simpler to cherry-pick your changes over to a new branch that is off develop.

**Pull request closed without merging**